### PR TITLE
r/glue_ml_transform - new resource

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_glue_ml_transform - add new resource
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSGlueMLTransform_'
--- PASS: TestAccAWSGlueMLTransform_description (127.24s)
--- PASS: TestAccAWSGlueMLTransform_glueVersion (135.07s)
--- PASS: TestAccAWSGlueMLTransform_maxRetries (170.36s)
--- PASS: TestAccAWSGlueMLTransform_tags (166.78s)
--- PASS: TestAccAWSGlueMLTransform_timeout (135.02s)
--- PASS: TestAccAWSGlueMLTransform_maxCapacity (166.63s)
--- PASS: TestAccAWSGlueMLTransform_disappears (83.80s)
--- PASS: TestAccAWSGlueMLTransform_workerType (97.77s)
--- PASS: TestAccAWSGlueMLTransform_typeFindMatchesFull (128.74s)
```

### CODE REVIEW

File: website/docs/r/glue_ml_transform.html.markdown [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+The following arguments are supported:
+
+* `name` – (Required) The name you assign to this ML Transform. It must be unique in your account.
+* `input_record_tables` - (Required)  A list of AWS Glue table definitions used by the transform. see [Input Record Tables](#input_record_tables).
+* `parameters` - (Required) The algorithmic parameters that are specific to the transform type used. Conditionally dependent on the transform type. see [Parameters](#parameters).
+* `role_arn` – (Required) The ARN of the IAM role associated with this ML Transform.
+* `description` – (Optional) Description of the ML Transform.
+* `glue_version` - (Optional) The version of glue to use, for example "1.0". For information about available versions, see the [AWS Glue Release Notes](https://docs.aws.amazon.com/glue/latest/dg/release-notes.html).
+* `max_capacity` – (Optional) he number of AWS Glue data processing units (DPUs) that are allocated to task runs for this transform. You can allocate from `2` to `100` DPUs; the default is 10. `max_capacity` is a mutually exclusive option with `number_of_workers` and `worker_type`.
```
Comment (ewbankkit) [inline website/docs/r/glue_ml_transform.html.markdown:133]:
```suggestion
* `max_capacity` – (Optional) The number of AWS Glue data processing units (DPUs) that are allocated to task runs for this transform. You can allocate from `2` to `100` DPUs; the default is `10`. `max_capacity` is a mutually exclusive option with `number_of_workers` and `worker_type`.
```

File: website/docs/r/glue_ml_transform.html.markdown [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+### input_record_tables
+
+* `database_name` - (Required) A database name in the AWS Glue Data Catalog.
+* `table_name` - (Required) A table name in the AWS Glue Data Catalog.
+* `catalog_id` - (Optional) A unique identifier for the AWS Glue Data Catalog.
+* `connection_name`- (Optional) The name of the connection to the AWS Glue Data Catalog.
+
+### parameters
+
+* `tansform_type` - (Required) The type of machine learning transform. For information about the types of machine learning transforms, see [Creating Machine Learning Transforms](http://docs.aws.amazon.com/glue/latest/dg/add-job-machine-learning-transform.html).
```
Comment (ewbankkit) [inline website/docs/r/glue_ml_transform.html.markdown:149]:
```suggestion
* `transform_type` - (Required) The type of machine learning transform. For information about the types of machine learning transforms, see [Creating Machine Learning Transforms](http://docs.aws.amazon.com/glue/latest/dg/add-job-machine-learning-transform.html).
```

File: website/docs/r/glue_ml_transform.html.markdown [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+* `enforce_provided_labels` - (Optional) The value to switch on or off to force the output to match the provided labels from users.
+* `precision_recall_trade_off` - (Optional) The value selected when tuning your transform for a balance between precision and recall.
+* `primary_key_column_name` - (Optional) The name of a column that uniquely identifies rows in the source table.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `arn` - Amazon Resource Name (ARN) of Glue ML Transform.
+* `id` - ML Transform ID.
```
Comment (ewbankkit) [inline website/docs/r/glue_ml_transform.html.markdown:164]:
```suggestion
* `id` - Glue ML Transform ID.
```

File: aws/resource_aws_glue_ml_transform.go [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+		return fmt.Errorf("error reading Glue ML Transform (%s): %w", d.Id(), err)
+	}
+
+	if output == nil {
+		log.Printf("[WARN] Glue ML Transform (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	log.Printf("setting Glue ML Transform: %#v", output)
```
Comment (ewbankkit) [inline aws/resource_aws_glue_ml_transform.go:245]:
```suggestion
	log.Printf("[DEBUG] setting Glue ML Transform: %#v", output)
```

**Comment by ewbankkit** [comment]
@DrFaust92 Now some failing new/updated linters 😢.

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSGlueMLTransform_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueMLTransform_ -timeout 120m
=== RUN   TestAccAWSGlueMLTransform_basic
=== PAUSE TestAccAWSGlueMLTransform_basic
=== RUN   TestAccAWSGlueMLTransform_typeFindMatchesFull
=== PAUSE TestAccAWSGlueMLTransform_typeFindMatchesFull
=== RUN   TestAccAWSGlueMLTransform_description
=== PAUSE TestAccAWSGlueMLTransform_description
=== RUN   TestAccAWSGlueMLTransform_glueVersion
=== PAUSE TestAccAWSGlueMLTransform_glueVersion
=== RUN   TestAccAWSGlueMLTransform_maxRetries
=== PAUSE TestAccAWSGlueMLTransform_maxRetries
=== RUN   TestAccAWSGlueMLTransform_tags
=== PAUSE TestAccAWSGlueMLTransform_tags
=== RUN   TestAccAWSGlueMLTransform_timeout
=== PAUSE TestAccAWSGlueMLTransform_timeout
=== RUN   TestAccAWSGlueMLTransform_workerType
=== PAUSE TestAccAWSGlueMLTransform_workerType
=== RUN   TestAccAWSGlueMLTransform_maxCapacity
=== PAUSE TestAccAWSGlueMLTransform_maxCapacity
=== RUN   TestAccAWSGlueMLTransform_disappears
=== PAUSE TestAccAWSGlueMLTransform_disappears
=== CONT  TestAccAWSGlueMLTransform_basic
=== CONT  TestAccAWSGlueMLTransform_timeout
=== CONT  TestAccAWSGlueMLTransform_disappears
=== CONT  TestAccAWSGlueMLTransform_maxCapacity
=== CONT  TestAccAWSGlueMLTransform_workerType
=== CONT  TestAccAWSGlueMLTransform_glueVersion
=== CONT  TestAccAWSGlueMLTransform_tags
=== CONT  TestAccAWSGlueMLTransform_maxRetries
=== CONT  TestAccAWSGlueMLTransform_description
=== CONT  TestAccAWSGlueMLTransform_typeFindMatchesFull
=== CONT  TestAccAWSGlueMLTransform_disappears
    resource_aws_glue_ml_transform_test.go:436: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSGlueMLTransform_disappears (43.63s)
2020/09/24 10:29:06 [DEBUG] Waiting for state to become: []
--- PASS: TestAccAWSGlueMLTransform_basic (50.70s)
--- PASS: TestAccAWSGlueMLTransform_timeout (77.92s)
--- PASS: TestAccAWSGlueMLTransform_description (80.14s)
--- PASS: TestAccAWSGlueMLTransform_maxRetries (87.89s)
--- PASS: TestAccAWSGlueMLTransform_glueVersion (93.40s)
--- PASS: TestAccAWSGlueMLTransform_workerType (93.50s)
--- PASS: TestAccAWSGlueMLTransform_maxCapacity (93.59s)
--- PASS: TestAccAWSGlueMLTransform_tags (106.50s)
--- PASS: TestAccAWSGlueMLTransform_typeFindMatchesFull (113.44s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	113.495s
```

**Comment by breathingdust** [comment]
LGTM! 🚀  Thanks @DrFaust92 

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TESTARGS='-run=TestAccAWSGlueMLTransform_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueMLTransform_ -timeout 120m
=== RUN   TestAccAWSGlueMLTransform_basic
=== PAUSE TestAccAWSGlueMLTransform_basic
=== RUN   TestAccAWSGlueMLTransform_typeFindMatchesFull
=== PAUSE TestAccAWSGlueMLTransform_typeFindMatchesFull
=== RUN   TestAccAWSGlueMLTransform_description
=== PAUSE TestAccAWSGlueMLTransform_description
=== RUN   TestAccAWSGlueMLTransform_glueVersion
=== PAUSE TestAccAWSGlueMLTransform_glueVersion
=== RUN   TestAccAWSGlueMLTransform_maxRetries
=== PAUSE TestAccAWSGlueMLTransform_maxRetries
=== RUN   TestAccAWSGlueMLTransform_tags
=== PAUSE TestAccAWSGlueMLTransform_tags
=== RUN   TestAccAWSGlueMLTransform_timeout
=== PAUSE TestAccAWSGlueMLTransform_timeout
=== RUN   TestAccAWSGlueMLTransform_workerType
=== PAUSE TestAccAWSGlueMLTransform_workerType
=== RUN   TestAccAWSGlueMLTransform_maxCapacity
=== PAUSE TestAccAWSGlueMLTransform_maxCapacity
=== RUN   TestAccAWSGlueMLTransform_disappears
=== PAUSE TestAccAWSGlueMLTransform_disappears
=== CONT  TestAccAWSGlueMLTransform_basic
=== CONT  TestAccAWSGlueMLTransform_timeout
=== CONT  TestAccAWSGlueMLTransform_disappears
=== CONT  TestAccAWSGlueMLTransform_maxCapacity
=== CONT  TestAccAWSGlueMLTransform_workerType
=== CONT  TestAccAWSGlueMLTransform_glueVersion
=== CONT  TestAccAWSGlueMLTransform_tags
=== CONT  TestAccAWSGlueMLTransform_maxRetries
=== CONT  TestAccAWSGlueMLTransform_typeFindMatchesFull
=== CONT  TestAccAWSGlueMLTransform_description
=== CONT  TestAccAWSGlueMLTransform_disappears
    resource_aws_glue_ml_transform_test.go:436: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSGlueMLTransform_disappears (41.57s)
--- PASS: TestAccAWSGlueMLTransform_glueVersion (51.26s)
--- PASS: TestAccAWSGlueMLTransform_description (61.28s)
--- PASS: TestAccAWSGlueMLTransform_basic (63.05s)
--- PASS: TestAccAWSGlueMLTransform_workerType (71.34s)
--- PASS: TestAccAWSGlueMLTransform_maxCapacity (75.55s)
--- PASS: TestAccAWSGlueMLTransform_timeout (75.60s)
--- PASS: TestAccAWSGlueMLTransform_maxRetries (78.39s)
--- PASS: TestAccAWSGlueMLTransform_tags (89.42s)
--- PASS: TestAccAWSGlueMLTransform_typeFindMatchesFull (135.04s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	136.423s

```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TESTARGS='-run=TestAccAWSGlueMLTransform_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueMLTransform_ -timeout 120m
=== RUN   TestAccAWSGlueMLTransform_basic
=== PAUSE TestAccAWSGlueMLTransform_basic
=== RUN   TestAccAWSGlueMLTransform_typeFindMatchesFull
=== PAUSE TestAccAWSGlueMLTransform_typeFindMatchesFull
=== RUN   TestAccAWSGlueMLTransform_description
=== PAUSE TestAccAWSGlueMLTransform_description
=== RUN   TestAccAWSGlueMLTransform_glueVersion
=== PAUSE TestAccAWSGlueMLTransform_glueVersion
=== RUN   TestAccAWSGlueMLTransform_maxRetries
=== PAUSE TestAccAWSGlueMLTransform_maxRetries
=== RUN   TestAccAWSGlueMLTransform_tags
=== PAUSE TestAccAWSGlueMLTransform_tags
=== RUN   TestAccAWSGlueMLTransform_timeout
=== PAUSE TestAccAWSGlueMLTransform_timeout
=== RUN   TestAccAWSGlueMLTransform_workerType
=== PAUSE TestAccAWSGlueMLTransform_workerType
=== RUN   TestAccAWSGlueMLTransform_maxCapacity
=== PAUSE TestAccAWSGlueMLTransform_maxCapacity
=== RUN   TestAccAWSGlueMLTransform_disappears
=== PAUSE TestAccAWSGlueMLTransform_disappears
=== CONT  TestAccAWSGlueMLTransform_basic
=== CONT  TestAccAWSGlueMLTransform_timeout
=== CONT  TestAccAWSGlueMLTransform_description
=== CONT  TestAccAWSGlueMLTransform_glueVersion
=== CONT  TestAccAWSGlueMLTransform_tags
=== CONT  TestAccAWSGlueMLTransform_maxRetries
=== CONT  TestAccAWSGlueMLTransform_disappears
=== CONT  TestAccAWSGlueMLTransform_maxCapacity
=== CONT  TestAccAWSGlueMLTransform_workerType
=== CONT  TestAccAWSGlueMLTransform_typeFindMatchesFull
2020/09/29 14:55:31 [DEBUG] Trying to get account information via sts:GetCallerIdentity
=== CONT  TestAccAWSGlueMLTransform_disappears
    resource_aws_glue_ml_transform_test.go:436: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSGlueMLTransform_disappears (22.09s)
--- PASS: TestAccAWSGlueMLTransform_basic (33.17s)
--- PASS: TestAccAWSGlueMLTransform_maxRetries (42.33s)
--- PASS: TestAccAWSGlueMLTransform_description (52.32s)
--- PASS: TestAccAWSGlueMLTransform_workerType (56.03s)
--- PASS: TestAccAWSGlueMLTransform_glueVersion (59.38s)
--- PASS: TestAccAWSGlueMLTransform_typeFindMatchesFull (67.03s)
--- PASS: TestAccAWSGlueMLTransform_timeout (80.34s)
--- PASS: TestAccAWSGlueMLTransform_tags (83.12s)
--- PASS: TestAccAWSGlueMLTransform_maxCapacity (138.63s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	140.014s

```

**Pull request merged**